diesel/src/associations/belongs_to.rs

```diff
@@ -27,8 +27,9 @@ impl<Parent, Child, Iter> GroupedBy<Parent> for Iter where
         let id_indices: HashMap<_, _> = parents.iter().enumerate().map(|(i, u)| (u.id(), i)).collect();
```

Comment:
Actually, what put me off at first about this was the HashMap—is this really necessary? I guess so, and I don't see a better way. Maybe use a hasher like Fnv which is optimised for smaller keys? Downsides might be that it's (a) not worth the effort (though I think this gets called quite a lot for each association), and (b) that this is not a cryptographic hasher, so it might be a security risk if users can influence the `id`s to do a collision attack.

Yes, I'm totally overthinking this.